bootloader/src/pyi_path.c

```diff
@@ -254,6 +256,11 @@ int pyi_path_executable(char *execfile, const char *appname)
 #elif defined(__sun)
     result = readlink("/proc/self/path/a.out", execfile, PATH_MAX);  // Solaris
 #endif
+    if(result >= 0) {
+        /* execfile is not yet zero-terminated. result is the byte count. */
+        *(execfile + result) = '\0';
+    }
+    else
     if(-1 == result) {
```

Comment:
So if C does no have an `elif`, can you please put he else and the if on the same line - if C permits. Otherwise it *looks* broken (while it is not)